### PR TITLE
add-tests

### PR DESCRIPTION
**Pull request by TheRealHaui**
Added new Unit Tests.

### CODE REVIEW

**Comment by TheRealHaui** [comment]
@dguy 
Would you mind merging this pull request too?

File: ksql-engine/src/test/java/io/confluent/ksql/util/StringUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+/**
+ * Unit tests for class {@link StringUtil}.
+ *
+ * @see StringUtil
+ */
+public class StringUtilTest {
+
+  @Test
+  public void testCleanQuotesWithNonEmptyString() {
```
Comment (dguy) [inline ksql-engine/src/test/java/io/confluent/ksql/util/StringUtilTest.java]: For these tests we probably should have `@Test(expected=KsqlException.class)`
Also, I think the naming could be more descriptive, i.e., what should it do?
something like `shouldThrowKsqlExceptionWhenQuotesNotClosed`

File: ksql-parser/src/test/java/io/confluent/ksql/parser/tree/CastTest.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+  @Test
+  public void testEqualsReturningTrue() {
+    ArithmeticBinaryExpression.Type arithmeticBinaryExpression_Type = ArithmeticBinaryExpression.Type.MODULUS;
+    ArithmeticBinaryExpression arithmeticBinaryExpression = new ArithmeticBinaryExpression(arithmeticBinaryExpression_Type, null, null);
+    Cast castOne = new Cast(arithmeticBinaryExpression, "pF(3kL7/byV6r0*W7@0", true);
+    NodeLocation nodeLocation = new NodeLocation((-1862), (-1862));
+    Cast castTwo = new Cast(nodeLocation, arithmeticBinaryExpression, "pF(3kL7/byV6r0*W7@0", true);
+
+    assertTrue(castOne.equals(castOne));
```
Comment (dguy) [inline ksql-parser/src/test/java/io/confluent/ksql/parser/tree/CastTest.java]: `assertThat(castOne, equalTo(castOne))`
same elsewhere

File: ksql-parser/src/test/java/io/confluent/ksql/parser/tree/CastTest.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+  @Test
+  public void testEqualsReturningFalse() {
+    ArithmeticBinaryExpression.Type arithmeticBinaryExpression_Type = ArithmeticBinaryExpression.Type.MODULUS;
+    ArithmeticBinaryExpression arithmeticBinaryExpression = new ArithmeticBinaryExpression(arithmeticBinaryExpression_Type, null, null);
+    NodeLocation nodeLocation = new NodeLocation((-1862), (-1862));
+    Cast castOne = new Cast(nodeLocation, arithmeticBinaryExpression, "^tgVV", true);
+    Cast castTwo = new Cast(arithmeticBinaryExpression, "^tgVV");
+
+    assertFalse(castOne.equals(castTwo));
```
Comment (dguy) [inline ksql-parser/src/test/java/io/confluent/ksql/parser/tree/CastTest.java]: `assertThat(castOne, not(equalTo(castTwo))`

File: ksql-parser/src/test/java/io/confluent/ksql/parser/tree/ExtractTest.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+public class ExtractTest {
+
+  @Test
+  public void testEqualsReturningTrue() {
+    QualifiedNameReference qualifiedNameReference = new QualifiedNameReference(null);
+    Extract.Field extract_Field = Extract.Field.MONTH;
+    Extract extractOne = new Extract(qualifiedNameReference, extract_Field);
+    Extract extractTwo = new Extract(qualifiedNameReference, extract_Field);
+
+    assertTrue(extractOne.equals(extractTwo));
```
Comment (dguy) [inline ksql-parser/src/test/java/io/confluent/ksql/parser/tree/ExtractTest.java]: as above

File: ksql-parser/src/test/java/io/confluent/ksql/parser/tree/CastTest.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    assertFalse(castOne.equals(null));
```
Comment (dguy) [inline ksql-parser/src/test/java/io/confluent/ksql/parser/tree/CastTest.java]: I'd probably put these into a different tests, i.e, `shouldNotBeEqualWhenNull` and `shouldNotBeEqualWhenNotCastType`

File: ksql-parser/src/test/java/io/confluent/ksql/parser/tree/ExtractTest.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    assertTrue(extractOne.equals(extractOne));
+  }
+
+  @Test
+  public void testEqualsWithNull() {
```
Comment (dguy) [inline ksql-parser/src/test/java/io/confluent/ksql/parser/tree/ExtractTest.java]: `shouldNotBeEqualWithNull`

File: ksql-parser/src/test/java/io/confluent/ksql/parser/tree/ExtractTest.java [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+  public void testEqualsReturningFalse() {
+    NullLiteral nullLiteral = new NullLiteral();
+    NullIfExpression nullIfExpression = new NullIfExpression(nullLiteral, nullLiteral);
+    Extract.Field extract_Field = Extract.Field.DAY_OF_WEEK;
+    Extract extractOne = new Extract(nullIfExpression, extract_Field);
+    NodeLocation nodeLocation = new NodeLocation(1513, 1513);
+    Extract extractTwo = new Extract(nodeLocation, nullLiteral, extract_Field);
+
+    assertFalse(extractOne.equals(extractTwo));
+    assertFalse(extractOne.equals(new Object()));
```
Comment (dguy) [inline ksql-parser/src/test/java/io/confluent/ksql/parser/tree/ExtractTest.java]: different test for this, as above.

File: ksql-parser/src/test/java/io/confluent/ksql/parser/tree/IntersectTest.java [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+import static org.junit.Assert.fail;
+
+/**
+ * Unit tests for class {@link Intersect}.
+ *
+ * @see Intersect
+ */
+public class IntersectTest {
+
+  @Test
```
Comment (dguy) [inline ksql-parser/src/test/java/io/confluent/ksql/parser/tree/IntersectTest.java]: can just use `@Test(expected=NullPointerException.class)`

File: ksql-parser/src/test/java/io/confluent/ksql/parser/tree/IntersectTest.java [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+  public void testFailsToCreateIntersectTakingNullAsFirstArgument() {
```
Comment (dguy) [inline ksql-parser/src/test/java/io/confluent/ksql/parser/tree/IntersectTest.java]: `shouldThrowExceptionWhenRelationsIsNull` ?

File: ksql-parser/src/test/java/io/confluent/ksql/parser/tree/SubqueryExpressionTest.java [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+ * @see SubqueryExpression
+ */
+public class SubqueryExpressionTest {
+
+  @Test
+  public void testEqualsReturningTrue() {
+    SubqueryExpression subqueryExpressionOne = new SubqueryExpression(null);
+    SubqueryExpression subqueryExpressionTwo = new SubqueryExpression(null);
+
+    assertTrue(subqueryExpressionOne.equals(subqueryExpressionTwo));
```
Comment (dguy) [inline ksql-parser/src/test/java/io/confluent/ksql/parser/tree/SubqueryExpressionTest.java]: as above `assertThat(...)`

File: ksql-parser/src/test/java/io/confluent/ksql/parser/tree/SymbolReferenceTest.java [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+ * @see SymbolReference
+ */
+public class SymbolReferenceTest {
+
+  @Test
+  public void testEqualsReturningTrue() {
+    SymbolReference symbolReferenceOne = new SymbolReference("");
+    SymbolReference symbolReferenceTwo = new SymbolReference("");
+
+    assertTrue(symbolReferenceOne.equals(symbolReferenceTwo));
```
Comment (dguy) [inline ksql-parser/src/test/java/io/confluent/ksql/parser/tree/SymbolReferenceTest.java]: ditto

**Pull request closed without merging**